Build on Linux sparc64 fails with #error "Undefined Architecture."

The build of abseil fails on Linux sparc64 with:

```
[ 69%] Building CXX object absl/debugging/CMakeFiles/examine_stack.dir/internal/examine_stack.cc.o
cd "/<<PKGBUILDDIR>>/static/absl/debugging" && /usr/bin/c++  -I"/<<PKGBUILDDIR>>" -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -Wall -Wextra -Wcast-qual -Wconversion-null -Wmissing-declarations -Woverlength-strings -Wpointer-arith -Wundef -Wunused-local-typedefs -Wunused-result -Wvarargs -Wvla -Wwrite-strings -Wno-missing-field-initializers -Wno-sign-compare -DNOMINMAX -std=gnu++14 -o CMakeFiles/examine_stack.dir/internal/examine_stack.cc.o -c "/<<PKGBUILDDIR>>/absl/debugging/internal/examine_stack.cc"
/<<PKGBUILDDIR>>/absl/debugging/internal/examine_stack.cc:72:2: error: #error "Undefined Architecture."
   72 | #error "Undefined Architecture."
      |  ^~~~~
[ 70%] Linking CXX static library libabsl_time.a
cd "/<<PKGBUILDDIR>>/static/absl/time" && /usr/bin/cmake -P CMakeFiles/time.dir/cmake_clean_target.cmake
cd "/<<PKGBUILDDIR>>/static/absl/time" && /usr/bin/cmake -E cmake_link_script CMakeFiles/time.dir/link.txt --verbose=1
/usr/bin/ar qc libabsl_time.a CMakeFiles/time.dir/civil_time.cc.o CMakeFiles/time.dir/clock.cc.o CMakeFiles/time.dir/duration.cc.o CMakeFiles/time.dir/format.cc.o CMakeFiles/time.dir/time.cc.o
/usr/bin/ranlib libabsl_time.a
make[4]: Leaving directory '/<<PKGBUILDDIR>>/static'
[ 70%] Built target time
```

Full build log in: https://buildd.debian.org/status/fetch.php?pkg=abseil&arch=sparc64&ver=0%7E20200923.3-1&stamp=1612822447&raw=0

A sparc64 porterbox running Debian unstable can be access through the gcc compile farm:

> https://gcc.gnu.org/wiki/CompileFarm

--- a/absl/base/internal/direct_mmap.h
+++ b/absl/base/internal/direct_mmap.h
@@ -77,7 +77,8 @@ inline void* DirectMmap(void* start, size_t length, int prot, int flags, int fd,
     (defined(__mips__) && _MIPS_SIM == _MIPS_SIM_ABI32) ||                   \
     (defined(__PPC__) && !defined(__PPC64__)) ||                             \
     (defined(__riscv) && __riscv_xlen == 32) ||                              \
-    (defined(__s390__) && !defined(__s390x__))
+    (defined(__s390__) && !defined(__s390x__)) ||                            \
+    (defined(__sparc__) && !defined(__arch64__))
   // On these architectures, implement mmap with mmap2.
   static int pagesize = 0;
   if (pagesize == 0) {

--- a/absl/debugging/internal/examine_stack.cc
+++ b/absl/debugging/internal/examine_stack.cc
@@ -63,6 +63,10 @@ void* GetProgramCounter(void* vuc) {
     return reinterpret_cast<void*>(context->uc_mcontext.psw.addr & 0x7fffffff);
 #elif defined(__s390__) && defined(__s390x__)
     return reinterpret_cast<void*>(context->uc_mcontext.psw.addr);
+#elif defined(__sparc__) && !defined(__arch64__)
+    return reinterpret_cast<void*>(context->uc_mcontext.gregs[19]);
+#elif defined(__sparc__) && defined(__arch64__)
+    return reinterpret_cast<void*>(context->uc_mcontext.mc_gregs[19]);
 #elif defined(__x86_64__)
     if (16 < ABSL_ARRAYSIZE(context->uc_mcontext.gregs))
       return reinterpret_cast<void*>(context->uc_mcontext.gregs[16]);

